src/dataframe.rs

```diff
@@ -38,8 +38,8 @@ pub(crate) struct PyDataFrame {
 
 impl PyDataFrame {
     /// creates a new PyDataFrame
-    pub fn new(df: Arc<DataFrame>) -> Self {
-        Self { df }
+    pub fn new(df: DataFrame) -> Self {
+        Self { df: Arc::new(df) }
```

Comment:
Pyo3 methods require `&self` and cannot take `self`, so my solution was to introduce an `Arc` here.